server/src/test/java/org/elasticsearch/search/suggest/SuggestTests.java

```diff
@@ -248,6 +247,22 @@ public void testSerialization() throws IOException {
         );
 
         final Suggest suggest = createTestItem();
+        // if bwc version is below V_8_8_0 we need to make sure potential
+        // search hits in CompletionSuggestion$Entry$Option don't have "rank" set
+        // TODO remove this once MINIMUM_COMPATIBLE > V_8_8_0
+        if (bwcVersion.before(TransportVersions.V_8_8_0)) {
```

Comment:
Does it make sense to change this TODO into an assertion, so that when we upgrade it will fail CI and prompt whoever is doing it to remove the block?